### ISSUE TITLE
Providing `null` as a value to a query causes an error

### ISSUE THREAD

**Issue by Jiralite** [mariadb-corporation/mariadb-connector-nodejs#167]
## Problem

Hiya, I migrated from another package to this one (this looks great by the way) and found some peculiar behaviour. My query is like this:

```TypeScript
<Pool>.query("UPDATE `table` SET `column` = ?;", null);
```

This would set the column to `null`. However, it doesn't work and I receive this:

```JavaScript
{
  text: "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '?' at line 1",
  sql: 'UPDATE `table` SET `column` = ?; - parameters:[]',
  fatal: false,
  errno: 1064,
  sqlState: '42000',
  code: 'ER_PARSE_ERROR'
}
```

The [documentation](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/documentation/promise-api.md#poolquerysql-values---promise) states that `values` is an array or an object **but in the case of one placeholder, it can be given as is**. In this case, I gave it as is, and it failed.

It seems to only happen with `null`. I assume this is a bug but a low priority one? This is fixed if I simply provide an array with `null` in it. Alas, here is my issue, please enlighten me if this is a bug or not!

## Versions
- Node.js: 16.6.1
- mariadb: 2.5.4

**Comment by rusher** [comment]
This is clearly a bug. 
null must here be considered as a value. 
I've created https://jira.mariadb.org/browse/CONJS-173 for that

**Comment by rusher** [comment]
corrected with https://github.com/mariadb-corporation/mariadb-connector-nodejs/commit/016ef46a672346d50a4018fea260ba7144f0ade9.
Will be available in 2.5.5 and 3.0.1-RC
